packages/ui/src/RadioGroup.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { ChangeEvent, FC, ReactNode, useRef } from 'react'
+import { v4 as uuid } from 'uuid'
+import { Error, errorId } from './Error'
+import classNames from 'classnames'
+import styles from './RadioGroup.module.scss'
+import { DataTestProp } from '@hazelcast/helpers'
+import { RadioGroupContext } from './RadioGroupContext'
+
+export type RadioGroupCoreProps = {
+  name: string
+  children: ReactNode | Array<ReactNode>
+  inline?: boolean
+}
+
+export type RadioGroupProps = RadioGroupCoreProps & {
+  onChange: (e: ChangeEvent<HTMLInputElement>) => void
+  radioGroupClassName?: string
+  error?: string
+  className?: string
+} & DataTestProp
+
+/**
+ * ### Purpose
+ * Forms require input from users. If you need information that can be represented as a choice of multiple values, use input with type 'radio'.
+ * You can use RadioGroup in order to group radios buttons into logical cells and provide corresponding error if needed.
+ */
+export const RadioGroup: FC<RadioGroupProps> = ({
+  radioGroupClassName,
+  error,
+  children,
+  className,
+  name,
+  onChange,
+  inline = false,
+  'data-test': dataTest,
+}) => {
+  const idRef = useRef(uuid())
+
+  return (
+    <div className={className} data-test={dataTest}>
+      <div
+        role="radiogroup"
+        data-test="radio-group"
+        className={classNames(
+          {
+            [styles.inline]: inline,
+          },
+          [styles.radioGroup, radioGroupClassName],
+        )}
+        aria-invalid={!!error}
+        aria-errormessage={error && errorId(idRef.current)}
```

Comment:
On Ubuntu the default screen reader does not read the error message out loud if it is linked to the radio group. Could you check it on Mac?

I would suggest passing `errorId?: string` to the context. Then we would be able to apply `aria-invlid` and `aria-errormessage` to every radio. Maybe, it would help with the screen reader. WDYT?

Reply:
It reads it for me, what I could do is to pass errorId instead of error message right now and apply to only for checked button?